cms/templates/container.html

```diff
@@ -12,7 +12,7 @@
 from django.utils.translation import ugettext as _
 from openedx.core.lib.js_utils import escape_json_dumps
 %>
-<%block name="title">${xblock.display_name_with_default} ${xblock_type_display_name(xblock) | h}</%block>
+<%block name="title">${xblock.display_name_unescaped} ${xblock_type_display_name(xblock) | h}</%block>
```

Comment:
This is producing a string that is half unescaped (display_name_unescaped) and half escaped (xblock_type_display_name + h filter).  These two need to stay in sync so that the place where this gets used (base.html) can either escape or not.

Ultimately, I think we want the following (assumes a future state where display_name_with_default were not escaped):
```
<%block name="title">${xblock.display_name_with_default | h} ${xblock_type_display_name(xblock) | h}</%block>
```
For now, we would either want to keep the original line (since it is already escaped), or switch to some form of the following, depending on what this method is ultimately named:
<%block name="title">${xblock.display_name_unescaped | h} ${xblock_type_display_name(xblock) | h}</%block>
 
Continue reading for a longer discussion about our options and the above proposal...

If we did switch this block to be unescaped (i.e. by dropping the existing h filter), it would require escaping in base.html where it is used.  The code in base.html would have to change from:
```
<title>
        <%block name="title"></%block> |
       ...
</title>
```
to: 
```
<title>
        <%block name="title" filter="h"></%block> |
       ...
</title>
Note: I'm new to Mako and inheritance, so I can't guarantee that this would work, but we would need something of this sort.

This %block represents a tricky situation because we have proposed escaping late where we see the context.  Or, if we must escape early we are considering using a naming convention like using "title_html" in place of "title" to designate we have html and not plain text.

However, I'm going to propose that we consider the use of a %block as "late" where we know the context.  Most of the time a %block will include tags that do in fact provide the context.  I think we should assume that %block's that contain just a string be considered html and that their contents should be html escaped, even if it doesn't contain the enclosing tag that makes it clear it is html.  If it is a block with just a string that will be used in a non-html context (e.g. javascript), the name could be used to indicate as such, and again the contents should already be javascript escaped.  I think this will be the simplest way of keeping us out of trouble, but this is just a proposal.  The reason I propose this is that most %block's do contain html, and they don't use a name like "something_html", so adding the "html" ending to only certain blocks wouldn't be meaningful enough.  @nedbat Thoughts?